src/main/kotlin/no/nav/familie/ef/sak/vurdering/medlemskap/MedlemskapRegelsett.kt

```diff
@@ -0,0 +1,246 @@
+
+    private val medlemskapslengdeAvbruddMerEnn10år =
+            Spesifikasjon<Medlemskapsgrunnlag>(beskrivelse = "Har søker etter fylte 16 år vært medlem i minst 7 år " +
+                                                             "ved avbrudd mer enn 10 år?",
+                                               identifikator = "EF-M10",
+                                               implementasjon = { medlemskapslengdeAvbruddMerEnn10år(this) })
+
+    private val nordiskStatsborger =
+            Spesifikasjon<Medlemskapsgrunnlag>(beskrivelse = "Er søker norsk/nordisk statsborger?",
+                                               identifikator = "EF-M8/17",
+                                               implementasjon = { nordiskStatsborger(this) })
+
+
+    private val eøsBorgerEllerLovligOpphold =
+            Spesifikasjon<Medlemskapsgrunnlag>(beskrivelse = "Har søker lovlig opphold i Norge eller " +
+                                                             "er søker EØS-borger med oppholdsrett?",
+                                               identifikator = "EF-M12/18",
+                                               implementasjon = { eøsBorgerEllerLovligOpphold(this) })
+
+    private val oppholdsrett = (nordiskStatsborger eller eøsBorgerEllerLovligOpphold)
+            .med("FP_VK 2.12", "Har søker lovlig opphold i Norge eller er søker EØS-borger med oppholdsrett")
+
+    private val forutgåendeMedlemskapSøker =
+            (medlemskapslengde eller medlemskapslengdeAvbruddMindreEnn10år eller medlemskapslengdeAvbruddMerEnn10år)
+                    .med("EF-M7 EF-M8 EF-M10", "Er vilkår for forutgående medlemskap oppfylt?")
+
+    val vurderingMedlemskapSøker =
+            (erMedlem
+                    og erBosattINorge
+                    og (oppholdINorge eller utenlandsoppholdGrunnetNorskArbeidsgiver)
+                    og forutgåendeMedlemskapSøker
+                    og oppholdsrett).med("Medlemskap kap. 15",
+                                         "Har søker opphold i Norge og forutgående medlemskap?")
+
+    private fun erMedlem(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        val statusNå =
+                medlemskapsgrunnlag.medlemskapshistorikk.medlemskapsperioder.find { it.inneholder(LocalDate.now()) }
+        if (statusNå == null || statusNå.gyldig == false) {
+            return Evaluering(Resultat.NEI, "Er IKKE MEDLEM")
+        }
+        if (statusNå.gyldig == null) {
+            return Evaluering(Resultat.KANSKJE, "Uavklart medlemskapsunntak nå")
+        }
+        return Evaluering.ja("Søke er registrert som medlem nå.")
+    }
+
+    private fun erBosattINorge(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        return if ("bosatt" == medlemskapsgrunnlag.søker.person.folkeregisterpersonstatus.firstOrNull()?.status) {
+            Evaluering(Resultat.JA, "Bosatt i Norge.")
```

Comment:
Er `Evaluering.ja` og `Evaluering(RESULTAT.JA` det samme? Isåfall Ser `Evaluering.ja` ut til å funke bra.

Reply:
Det er det samme. Bare jeg som har vært litt dårlig til å være konsistent. Skal fikse det.